### PR TITLE
fix: resolve various parser issues

### LINKED ISSUE #2: Missing esx.register of imported component
There are no `esx.register` call when importing component with `import` statement.

```javascript
test('JSX component registration – import', async ({ is }) => {
  const src = [
    `import React from 'react'`,
    `import Foo from './foo'`,
    'module.exports = <div><Foo/></div>'
  ].join('\n')
  const esx = [
    `const esx = require('esx')()`,
    `import React from 'react'`,
    `import Foo from './foo'`,
    'esx.register({ Foo })',
    'module.exports = esx `<div><Foo/></div>`'
  ].join('\n')
  is(convert(src), esx)
})
```

```diff
--- wanted                                
+++ found                                 
const esx = require('esx')()             
import React from 'react'                
import Foo from './foo'                  
-esx.register({ Foo })                    
module.exports = esx `<div><Foo/></div>`
```

### PR DESCRIPTION
**Pull request by irudoy**
* set `ecmaVersion` acorn option to `2020` (https://github.com/esxjs/esx/issues/17)
* fix function attributes parsing (https://github.com/esxjs/esx/issues/17)
* fix imported components registration (fix: #2)
* fix computed props
* add lockfile

### CODE REVIEW

**Comment by prograhammer** [comment]
@irudoy I tried out this change (just directly added the code in node_modules to-esx) and still see it breaking on optional chaining (via babel-plugin-esx-ssr).

**Comment by irudoy** [comment]
@prograhammer can you provide stack trace of error thrown with these changes?

**Comment by prograhammer** [comment]
@irudoy there's no new errors that appear with these changes, I'm just noticing that optional chaining still breaks. See here: https://github.com/esxjs/esx/issues/17

**Comment by prograhammer** [comment]
@irudoy BTW, I'm able to get through this issue by transpiling (via babel-loader) twice. I do one pass through babel on the optional chaining, etc, then another pass with `babel-plugin-esx-ssr`. However now I'm facing an issue with none of the components registering from esx:

```js
ReferenceError: ESX: FooComponent not found in registered components","stack":"ReferenceError: ESX: 
FooComponent not found in registered components\n    at grow (webpack-
internal:///./node_modules/esx/lib/parse.js:116:13)\n    at parse (webpack-
internal:///./node_modules/esx/lib/parse.js:363:11)\n    at raw (webpack-internal:///./node_modules/esx/index.js:292:24)\n    
at render (webpack-internal:///./node_modules/esx/index.js:350:31)\n 
```
I get the above error with or without your changes.  

Maybe related to this: https://github.com/esxjs/to-esx/issues/2

**Comment by prograhammer** [comment]
@irudoy Update: I think the `ReferenceError: ESX: FooComponent not found in registered components` error my be comiing from `renderToNodeStream`. I tried with`renderToString` and the error went away. I still need to confirm ESX is working though.

**Comment by prograhammer** [comment]
@irudoy Now I'm transforming `import/exports` to commonjs (`require`). That get's me past the register errors, but now I'm getting `ReferenceError: React is not defined`.

**Comment by irudoy** [comment]
@prograhammer first of all consider, that esx project isn't stable for now, and I'm also trying to get it to work with my codebase as well 😄 Personally, I'm not using `babel-plugin-esx-ssr` because it has issues, and with `webpack` it's more rational to use a loader to convert the whole module to esx.

Registration issues with imports should be resolved by this PR. Could you create an isolated repro? And for `React is not defined` too. Seems like it's related to your build tool setup.

File: .gitignore [hunk 1]
```diff
@@ -46,10 +46,6 @@ profile-*
 # VS Code
 .vscode/
 
-# lock files
-yarn.lock
-package-lock.json
```
Comment (davidmarkclements) [inline .gitignore:51]: we do want to ignore lock files, imo lock files are for applications, not modules

**Review by davidmarkclements — CHANGES_REQUESTED** [review]
rm lock files (lock files do not belong in modules) 

I think if this causes another part to break, even if that other part is experimental we need to at least isolate the cause and rationalize it (if not fix it) before we can merge this

**Pull request closed without merging**